src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/detentionengagementteam/DetentionEngagementTeamRequestResponseReviewPersonalisation.java

```diff
@@ -0,0 +1,126 @@
+    private final CustomerServicesProvider customerServicesProvider;
+    private final String detentionEngagementTeamRequestResponseReviewTemplateId;
+    private final String iaExUiFrontendUrl;
+    private final String adaPrefix;
+    private final DetEmailService detEmailService;
+    private final DocumentDownloadClient documentDownloadClient;
+
+    public DetentionEngagementTeamRequestResponseReviewPersonalisation(
+        @Value("${govnotify.template.responseReviewDirection.detentionEngagementTeam.email}") String detentionEngagementTeamRequestResponseReviewTemplateId,
+        @Value("${iaExUiFrontendUrl}") String iaExUiFrontendUrl,
+        @Value("${govnotify.emailPrefix.ada}") String adaPrefix,
+        CustomerServicesProvider customerServicesProvider,
+        DetEmailService detEmailService,
+        DocumentDownloadClient documentDownloadClient
+    ) {
+        this.detentionEngagementTeamRequestResponseReviewTemplateId = detentionEngagementTeamRequestResponseReviewTemplateId;
+        this.iaExUiFrontendUrl = iaExUiFrontendUrl;
+        this.adaPrefix = adaPrefix;
+        this.customerServicesProvider = customerServicesProvider;
+        this.detEmailService = detEmailService;
+        this.documentDownloadClient = documentDownloadClient;
+    }
+
+    @Override
+    public String getTemplateId() {
+        return detentionEngagementTeamRequestResponseReviewTemplateId;
+    }
+
+    @Override
+    public Set<String> getRecipientsList(AsylumCase asylumCase) {
+        return Collections.singleton(detEmailService.getAdaDetEmailAddress());
+    }
+
+    @Override
+    public String getReferenceId(Long caseId) {
+        return caseId + "_DETENTION_ENGAGEMENT_TEAM_REQUEST_RESPONSE_REVIEW";
+    }
+
+    @Override
+    public Map<String, Object> getPersonalisationForLink(AsylumCase asylumCase) {
+        requireNonNull(asylumCase, "asylumCase must not be null");
+
+        String hearingDate = LocalDate.parse(asylumCase.read(LIST_CASE_HEARING_DATE, String.class)
+                .orElseThrow(() -> new IllegalStateException("Hearing date is not present")))
+            .format(DateTimeFormatter.ofPattern("d MMM yyyy"));
+
+        String ariaListingReferenceIfPresent = asylumCase.read(ARIA_LISTING_REFERENCE, String.class)
+            .map(ariaListingReference -> "Listing reference: " + ariaListingReference)
+            .orElse("");
+
+        String appealReviewOutcome = getAppealReviewOutcome(asylumCase);
+        String documentDownloadTitle = appealReviewOutcome.equals(WITHDRAW) ? WITHDRAWAL_LETTER : HOME_OFFICE_RESPONSE;
+
+        return ImmutableMap
+            .<String, Object>builder()
+            .putAll(customerServicesProvider.getCustomerServicesPersonalisation())
+            .put("subjectPrefix", adaPrefix)
+            .put("appealReferenceNumber", asylumCase.read(AsylumCaseDefinition.APPEAL_REFERENCE_NUMBER, String.class).orElse(""))
+            .put("ariaListingReferenceIfPresent", ariaListingReferenceIfPresent)
```

Comment:
If ariaListingReferenceIfPresent is empty, will this not be visible in the message?

Reply:
No it will not.